### PR TITLE
feat(PE-7174): support for primary name protocol with io process

### PR DESCRIPTION
**Pull request by dtfiedler**
These can be extended to allow controllers to create and remove primary name claims.

Related PR: https://github.com/ar-io/ar-io-network-process/pull/175

TODO:
- sending `Remove-Primary-Name` message to IO process when updating a name. Behind a flag like `Cast`?

_any state changes we want to store on an ANT that track primary names to owners they may care about?_

### CODE REVIEW

**Comment by atticusofsparta** [comment]
Will add tests and an empty handler for "Approve-Primary-Name-Request" with docs on automatic approval

File: src/common/main.lua [hunk 1]
```diff
@@ -262,13 +266,52 @@ function ant.init()
...
+			Action = "Reassign-Name-Notice",
 			Initiator = msg.From,
 			Name = name,
 			["Process-Id"] = antProcessIdToReassign,
 		})
 	end)
+
+	createActionHandler(ActionMap.ApproveName, function(msg)
+		--- NOTE: this could be modified to allow specific users/controllers to create claims
+		utils.validateOwner(msg.From)
```
Comment (dtfiedler) [inline src/common/main.lua:278]: @atticusofsparta - consider if we think allowing controllers to approve primary name requests is desired

Comment (atticusofsparta) [reply src/common/main.lua:278]: I have commenced developing one such module on personal projects but its still not mature.

**Review by fedellen — APPROVED** [review]
